modules/dnn/src/layers/cpu_kernels/softmax_kernels.simd.hpp

```diff
@@ -0,0 +1,386 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// This file is modified from the ficus (https://github.com/vpisarev/ficus/blob/master/lib/NN/OpNN.fx).
+// Here is the original license:
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+#include <opencv2/core/hal/intrin.hpp>
+#include <opencv2/core/utility.hpp> // parallel_for_
+
+#define _c_exp_hi 88.3762626647949f
+#define _c_exp_lo -88.3762626647949f
+
+#define _c_cephes_LOG2EF 1.44269504088896341
+#define _c_cephes_exp_C1 0.693359375
+#define _c_cephes_exp_C2 2.12194440e-4
+
+#define _c_cephes_exp_p0 1.9875691500E-4
+#define _c_cephes_exp_p1 1.3981999507E-3
+#define _c_cephes_exp_p2 8.3334519073E-3
+#define _c_cephes_exp_p3 4.1665795894E-2
+#define _c_cephes_exp_p4 1.6666665459E-1
+#define _c_cephes_exp_p5 5.0000001201E-1
```

Comment:
You can use `constexpr` in place of `#define`. See https://en.cppreference.com/w/cpp/language/constexpr.